File: classroom-booking/src/main/java/com/ilerna/classroom_booking/service/ClassroomTypeService.java
package com.ilerna.classroom_booking.service;

import java.util.List;

public interface ClassroomTypeService {
    ClassroomTypeService saveClassroomType(ClassroomTypeService classroomType);
    ClassroomTypeService getClassroomTypeById(Long id);
    List<ClassroomTypeService> getAllClassroomTypes();
    ClassroomTypeService updateClassroomType(ClassroomTypeService classroomType);
    void deleteClassroomTypeById(Long id);
}

File: classroom-booking-frontend/src/app/services/classroom-type.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ClassroomType } from '../models/classroom-type';


@Injectable({
  providedIn: 'root'
})

export class ClassroomTypeService {
  private baseUrl = 'http://localhost:8080/classroom-types';

  constructor(private http: HttpClient) { }

  createClassroomType(classroomType: ClassroomType): Observable<ClassroomType> {
    return this.http.post<ClassroomType>(this.baseUrl, classroomType);
  }

  getClassroomTypeById(id: number): Observable<ClassroomType> {
    return this.http.get<ClassroomType>(`${this.baseUrl}/${id}`);
  }

  getAllClassroomTypes(): Observable<ClassroomType[]> {
    return this.http.get<ClassroomType[]>(this.baseUrl);
  }

  updateClassroomType(id: number, classroomType: ClassroomType): Observable<ClassroomType> {
    return this.http.put<ClassroomType>(`${this.baseUrl}/${id}`, classroomType);
  }

  deleteClassroomType(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
